### ISSUE TITLE
Inability to place a colony/outpost should produce a sitrep

### ISSUE THREAD

**Issue by l29ah** [freeorion/freeorion#1456]
When an colony/outpost base is produced on a system where enemy is present and ordered to place a colony/outpost it won't get destroyed and silently ignore its orders. It should bug the player as one would easily forget about it.

**Comment by Dilvish-fo** [comment]
I don't think that this is something that could currently be handled via scripting a sitrep, and I think the best solution for this problem is not a sitrep, or at least not just a sitrep.  The constraint against colonizing when an armed aggressive enemy is present is enforced by c++ engine code in ServerApp.cpp,  HandleColonization(), not the scripts, and from a quick review it looks to me like the scripts currently do not have access to the info that a ship has been ordered to colonize (it looks to me like the only current similar type status that is exposed to scripts is the OrderedBombarded condition, but that exposing the same thing for colonization wouldn't really help here anyways).

I think that to cover this well we would need a couple things done:

1. Button Disabling:  The UI code that enables the colonize button should also check for armed aggressive enemies, and not enable itself in if they are present.  This button enablement is dealt with by [SidePanel::PlanetPanel::Refresh()](https://github.com/freeorion/freeorion/blob/9eab07614ec3c94441a3a3ae3f2d8d043799968f/UI/SidePanel.cpp#L1628-L1763), and it checks for a number of other things but for some reason doesn't check for enemy presence.
2. Failure SitRep:  The server checks are a bit more broad than the client can reliably check-- the client will only know about visible enemies, but stealthed armed aggressive enemies will also block colonization (and I think would not *necessarily* have been revealed by a previous round of combat), and also colonization will fail if multiple empires try to colonize the same planet at the same time.   I see several problems with trying to handle a sitrep via scripting, and I think that the best thing will be to generate a failure sitrep in [ServerApp.cpp  HandleColonization()](https://github.com/freeorion/freeorion/blob/9eab07614ec3c94441a3a3ae3f2d8d043799968f/server/ServerApp.cpp#L2232-L2233) the same way that the colonization [success sitrep is generated](https://github.com/freeorion/freeorion/blob/9eab07614ec3c94441a3a3ae3f2d8d043799968f/server/ServerApp.cpp#L2255-L2258).

**Comment by Oberlus** [comment]
I would like this to be fixed someday.
Either by having a SitRep or (better) by disabling the colonize/outpost button when there are enemies, as suggested by @Dilvish-fo
